### ISSUE TITLE
Support for Custom Functions?

### ISSUE THREAD

**Issue by abritinthebay** [mdarnall/hapi-sass#14]
It's a recent (but very powerful) addition to node-sass (and SASS in general).

Thoughts about how to include support?

**Comment by digitaldesigndj** [comment]
I don't use this project anymore TBH. I've found that keeping a separate build process for front-end stuff (say, gulp) helps structure my projects nicely. 

HAPI = API/Backend 
Gulp = Frontend Output

Hope this helps.

**Comment by mdarnall** [comment]
I haven't used the feature before. [Looks like](https://github.com/sass/node-sass#example) the `functions` option is passed to `render` as an object literal - Could we simply expose it as another option?

In your `server.js` you could then import your custom functions and pass them along to the hapi-sass when it's registered with the server. 

```js
let options = {
    src: './sass',
    dest: './css',
   functions: require('myCustomFunctions')
}

server.register([Inert, {
        register: HapiSass,
        options: options
    }]
    , function (err) { } 
)
```

**Comment by abritinthebay** [comment]
@digitaldesigndj - Gulp? Yikes. Gross ;) :P I *could* just use a script for it - we already have one for building for Production - but obviously that has limitations. We use Hapi-Sass mostly for Development mode as it's much more effective, faster, and more reliable, than a watch task.

@mdarnall - That's exactly how I'd imagine it would work, yes! I already have the`myCustomeFunctions.js` file so including it that way would be great.

**Comment by abritinthebay** [comment]
To speak to how useful this is: if you - like my company does - have a lot of different domains for things like static assets or CDN urls you may need to change urls depending on the environment you're running in (dev vs prod, for example).

Custom functions are a _really easy_ way to do this.

**Comment by mdarnall** [comment]
Gotcha, yeah I could see how that would be helpful. I think this would be pretty easy to add and I could probably get to it this week.

**Comment by abritinthebay** [comment]
That would be *amazing*.

**Comment by mdarnall** [comment]
@abritinthebay Take a look at https://github.com/mdarnall/hapi-sass/pull/15 and let me know what you think.

**Comment by abritinthebay** [comment]
Assuming that works for node-sass API (I tend to use it via CLI) then looks great to me 👍

**Comment by mdarnall** [comment]
Released as 2.3 🎉

**Comment by abritinthebay** [comment]
🍻
